src/Transloadit.ts

```diff
@@ -253,24 +288,33 @@ class TransloaditClient {
       }
 
       return Promise.race([createAssemblyAndUpload(), streamErrorPromise])
-    })()
+    })() as CreateAssemblyPromise
```

Comment:
could we instead do
```js
    const promise: CreateAssemblyPromise = (async () => {
```

Reply:
No. A promise is not assignable to `CreateAssemblyPromise`, but it can be cast to it. I’m not a fan of this pattern though. I wouldn’t expect a promise to have additional non-standard properties. I would rather get rid of the `CreateAssemblyPromise` type and return a regular promise. Do you know why this was done in the first place?

Reply:
It was [inspired by `got`](https://github.com/sindresorhus/got/blob/main/documentation/1-promise.md) which node-sdk uses. The reason why we put assemblyId on the promise directly is that sometimes one needs to get the assembly ID immediately before the promise is created/finished (before the promise fulfilled). in the next major we could probably change it if yo uhave some better ideas. one idea would be to return a an object with a `{ promise: Promise, assemblyId: string }`, but i'm not sure if it's any more clean

Reply:
I took the liberty of fixing this with Object.assign